ceph.spec.in

```diff
@@ -944,17 +944,15 @@ Group:		Development/Libraries/Python
 This package contains data structures, classes and functions used by Ceph.
 It also contains utilities used for the cephadm orchestrator.
 
-%if 0%{with cephfs_shell}
```

Comment:
but we still conditionalize this option like
```
%if 0%{?fedora} || 0%{?suse_version} >= 1500
# distros that ship cmd2 and/or colorama
%bcond_without cephfs_shell
%else
# distros that do _not_ ship cmd2/colorama
%bcond_with cephfs_shell
%endif
```

could you double check if this changed after we ditch the support of RHEL/CentOS 7?

if that's the case, we might want to remove them all together. otherwise user/maintainer might run into the issue where he/she gets `cephfs_shell` disabled but this packages is still built.

Reply:
with filter, without locale
---------------------------
code
```
        out, err = self.subproc.communicate()
        out, err = rm_nonascii_chars(out), rm_nonascii_chars(err)
        out, err = out.decode(), err.decode()
``` 

output
```
    2020-01-17 14:36:54,988.988 INFO:__main__:======================================================================
    2020-01-17 14:36:54,988.988 INFO:__main__:ERROR: test_put_and_get_without_target_directory (tasks.cephfs.test_cephfs_shell.TestGetAndPut)
    2020-01-17 14:36:54,988.988 INFO:__main__:----------------------------------------------------------------------
    2020-01-17 14:36:54,988.988 INFO:__main__:Traceback (most recent call last):
    2020-01-17 14:36:54,988.988 INFO:__main__:  File "/home/rishabh/repos/ceph/fs-shell-retval/qa/tasks/cephfs/test_cephfs_shell.py", line 186, in test_put_and_get_without_target_directory
    2020-01-17 14:36:54,988.988 INFO:__main__:    self.mount_a.run_shell(['mkdir', 'a'])
    2020-01-17 14:36:54,988.988 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 717, in run_shell
    2020-01-17 14:36:54,988.988 INFO:__main__:    omit_sudo=omit_sudo)
    2020-01-17 14:36:54,988.988 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 396, in run
    2020-01-17 14:36:54,988.988 INFO:__main__:    proc.wait()
    2020-01-17 14:36:54,988.988 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 196, in wait
    2020-01-17 14:36:54,988.988 INFO:__main__:    raise CommandFailedError(self.args, self.exitstatus)
    2020-01-17 14:36:54,988.988 INFO:__main__:CommandFailedError: Command failed with status 1: ['mkdir', 'a']
 ```

without filter, with locale
---------------------------
code
```
        import locale
        locale.setlocale(locale.LC_ALL, 'C')
        out, err = self.subproc.communicate()
        out, err = out.decode(), err.decode()
 ```

output
```
    2020-01-17 14:48:30,838.838 INFO:__main__:======================================================================
    2020-01-17 14:48:30,838.838 INFO:__main__:ERROR: test_put_and_get_without_target_directory (tasks.cephfs.test_cephfs_shell.TestGetAndPut)
    2020-01-17 14:48:30,838.838 INFO:__main__:----------------------------------------------------------------------
    2020-01-17 14:48:30,838.838 INFO:__main__:Traceback (most recent call last):
    2020-01-17 14:48:30,838.838 INFO:__main__:  File "/home/rishabh/repos/ceph/fs-shell-retval/qa/tasks/cephfs/test_cephfs_shell.py", line 186, in test_put_and_get_without_target_directory
    2020-01-17 14:48:30,838.838 INFO:__main__:    self.mount_a.run_shell(['mkdir', 'a'])
    2020-01-17 14:48:30,838.838 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 718, in run_shell
    2020-01-17 14:48:30,838.838 INFO:__main__:    omit_sudo=omit_sudo)
    2020-01-17 14:48:30,839.839 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 397, in run
    2020-01-17 14:48:30,839.839 INFO:__main__:    proc.wait()
    2020-01-17 14:48:30,839.839 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 186, in wait
    2020-01-17 14:48:30,839.839 INFO:__main__:    out, err = out.decode(), err.decode()
    2020-01-17 14:48:30,839.839 INFO:__main__:UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 31: ordinal not in range(128)
```